src/oni2-keyboard-layout/stubs/keyboard-layout.c

```diff
@@ -0,0 +1,23 @@
+#include <caml/alloc.h>
+#include <caml/callback.h>
+#include <caml/memory.h>
+#include <caml/mlvalues.h>
+#include <caml/threads.h>
+
+/* Create an OCaml Keymap Entry
+  Simple helper method to construct a keymap
+  entry from strings
+ */
+value createKeymapEntry(
+  const char *unmodified,
+  const char *withShift,
+  const char *withAltGraph,
+  const char *withAltGraphShift
+) {
+  value keymapEntry = caml_alloc(4, 0);
```

Comment:
I believe this will need to be wrapped with `CAMLparam0()`, `CAMLlocal1(keymapEntry)`, and `CAMLreturn(keymapEntry)`

Reply:
Fixed in 0641b93!